### PR TITLE
Fix the error when you go to the life bank information page outside the website

### LINKED ISSUE #284: Fix bug to open lifebank info page to the browse
when open lifebank info page to the browse this broke

- [x] modify the column to user table in BD
- [x] load profile based to usernameLink

### PR DESCRIPTION
**Pull request by AngeloCG97**
## Fix bug to open lifebank info page to the browse 

closes #284 

### What does this PR do?
- Create FK in location table whit user table.
- Change get URL direction
- Create query to get profile lifebank and modify other querry
- Modify MapPopup component

### How should this be tested?
- Run make run
- Go to localhost:3000
- Go to lifebank info page

### CODE REVIEW

**Review by JeanVegaD — CHANGES_REQUESTED** [review]
- fix Hasura migrations problem
- fix URL in the desktop version
- add not-found page when URL doesn't match

File: webapp/src/gql/account.gql.js [hunk 1]
```diff
@@ -102,7 +102,6 @@ export const EDIT_PROFILE_MUTATION = gql`
   }
 `
```
Comment (JustinCast) [inline webapp/src/gql/account.gql.js:104]: Remove this unnecessary white space please

File: webapp/src/routes/Home/ShowLifebanks.js [hunk 2]
```diff
@@ -190,11 +190,12 @@ const ShowLifebanks = ({ banks, loading, isDesktop }) => {
   }
 
   const BankItem = (props) => {
```
Comment (JustinCast) [inline webapp/src/routes/Home/ShowLifebanks.js:192]: You can use parenthesis when in the JSX Element body is just a return

File: webapp/src/routes/Home/ShowLifebanks.js [hunk 3]
```diff
@@ -190,11 +190,12 @@ const ShowLifebanks = ({ banks, loading, isDesktop }) => {
...
+
```
Comment (JustinCast) [inline webapp/src/routes/Home/ShowLifebanks.js:193]: Remove this white space please

**Review by JustinCast — CHANGES_REQUESTED** [review]

File: webapp/src/routes/InfoPage/InfoPage.js [hunk 4]
```diff
@@ -226,10 +230,33 @@ const InfoPage = () => {
...
+      username: window.location.pathname.slice(6, window.location.pathname.length)
+    },
+    skip: true
+  })
+
   useEffect(() => {
-    location.state
-      ? setProfile(location.state.profile)
-      : setProfile(location.state.profile)
+    if (location.state) {
```
Comment (JustinCast) [inline webapp/src/routes/InfoPage/InfoPage.js:241]: Remove these unnecessary curly braces

File: webapp/src/routes/InfoPage/InfoPage.js [hunk 5]
```diff
@@ -226,10 +230,33 @@ const InfoPage = () => {
...
+      setProfile(location.state.profile)
+    }
+    else {
+      const getProfile = async () => {
+        const { data } = await getData({
+          username: window.location.pathname.slice(6, window.location.pathname.length)
+        })
+
+        if (data.location.length > 0) {
+          setProfile(data.location[0])
```
Comment (JustinCast) [inline webapp/src/routes/InfoPage/InfoPage.js:251]: Use a ternary operator or remove the unnecessary curly braces

**Review by JustinCast — CHANGES_REQUESTED** [review]

**Comment by JustinCast** [comment]
image
There is a conflict here

File: webapp/src/routes/Home/ShowLifebanks.js [hunk 6]
```diff
@@ -189,39 +189,38 @@ const ShowLifebanks = ({ banks, loading, isDesktop }) => {
...
-        style={{ textDecoration: 'none' }}
-        to={{
-          pathname: 'info/' + props.bank.info.name.replaceAll(" ", "-"),
-          state: { profile: props.bank }
-        }}
+  const BankItem = (props) => (
+    <LinkRouter
+      style={{ textDecoration: 'none' }}
+      to={{
+        pathname: 'info/' + props.bank.user.username.replaceAll(" ", "-"),
```
Comment (tetogomez) [inline webapp/src/routes/Home/ShowLifebanks.js:196]: We can change this string concat (old school XD) and use template string.

example:
```
pathname: `info/${props.bank.user.username.replaceAll(" ", "-")}`
```

File: webapp/src/routes/Home/ShowLifebanks.js [hunk 7]
```diff
@@ -269,40 +268,38 @@ const ShowLifebanks = ({ banks, loading, isDesktop }) => {
...
+        </Box>
+      </Box>
+      <CardContent className={classes.cardContent}>
+        <Typography className={classes.cardContentText} >{truncateString(props.bank.info.description)}
+        </Typography>
+      </CardContent>
+      <LinkRouter
+        style={{ textDecoration: 'none' }}
+        to={{
+          pathname: 'info/' + props.bank.user.username.replaceAll(" ", "-"),
```
Comment (tetogomez) [inline webapp/src/routes/Home/ShowLifebanks.js:290]: same here, use template string

File: webapp/debug.log [hunk 8]
```diff
@@ -0,0 +1,2 @@
+[1014/085359.567:ERROR:directory_reader_win.cc(43)] FindFirstFile: El sistema no puede encontrar la ruta especificada. (0x3)
```
Comment (JustinCast) [inline webapp/debug.log:1]: What is this file @AngeloCG97 ?

File: webapp/src/components/MapPopup/index.js [hunk 9]
```diff
@@ -34,7 +34,7 @@ function MapPopup({ id, info }) {
       return "geo:0,0?q=" + info.geolocation.latitude + "," + info.geolocation.longitude
     }
     else {
-      return "http://maps.google.com/maps?q=" + info.geolocation.latitude + "," + info.geolocation.longitude
+      return "https://maps.google.com/maps?q=" + info.geolocation.latitude + "," + info.geolocation.longitude
```
Comment (JustinCast) [inline webapp/src/components/MapPopup/index.js:37]: use template string please

**Review by JustinCast — CHANGES_REQUESTED** [review]

**Review by JustinCast — APPROVED** [review]

**Review by JeanVegaD — APPROVED** [review]

**Pull request merged**